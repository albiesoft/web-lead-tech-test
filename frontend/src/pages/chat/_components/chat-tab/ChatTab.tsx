import { ChatForm } from "./_components/form/ChatForm";
import { MessagesList } from "./_components/messages/MessagesList";

const ChatTab = () => {
  return (
    <div className="flex-1 flex flex-col min-h-screen">
      <MessagesList />
      <ChatForm />
    </div>
  );
};

export default ChatTab;
