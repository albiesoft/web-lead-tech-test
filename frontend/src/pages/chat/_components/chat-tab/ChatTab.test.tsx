import SocketProvider from "@/context/socketContext";
import { render } from "@testing-library/react";
import ChatTab from "./ChatTab";

describe("<ChatTab />", () => {
  const wrapper = () => (
    <SocketProvider>
      <ChatTab />
    </SocketProvider>
  );

  it("should render correctly", () => {
    const comp = render(wrapper());

    expect(comp).toMatchSnapshot();
  });
});
