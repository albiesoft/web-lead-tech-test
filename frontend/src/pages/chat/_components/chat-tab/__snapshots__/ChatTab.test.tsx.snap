// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ChatTab /> > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex-1 flex flex-col min-h-screen"
      >
        <div
          class="flex-1 flex flex-col p-[5px] overflow-auto max-h-[490px]"
        >
          <div
            class="mt-auto"
          />
        </div>
        <div
          class="p-[20px] px-[10px]"
        >
          <form
            class="flex gap-[10px]"
            data-testid="chat-form"
          >
            <input
              class="flex-1 rounded-full border-[8px] border-[#cfcfcf] px-[12px] py-[8px]"
              data-testid="chat-form-input"
              placeholder="Message "
              type="text"
              value=""
            />
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex-1 flex flex-col min-h-screen"
    >
      <div
        class="flex-1 flex flex-col p-[5px] overflow-auto max-h-[490px]"
      >
        <div
          class="mt-auto"
        />
      </div>
      <div
        class="p-[20px] px-[10px]"
      >
        <form
          class="flex gap-[10px]"
          data-testid="chat-form"
        >
          <input
            class="flex-1 rounded-full border-[8px] border-[#cfcfcf] px-[12px] py-[8px]"
            data-testid="chat-form-input"
            placeholder="Message "
            type="text"
            value=""
          />
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
